src/xdr/Stellar-contract.x

```diff
@@ -0,0 +1,199 @@
+// Copyright 2022 Stellar Development Foundation and contributors. Licensed
```

Comment:
Is this XDR the XDR we should start using in rs-stellar-xdr?

Will this PR merge to master in the near or distant future?

Reply:
Yes I think this is the PR to track for XDR as we go forward.

I hope it'll merge soon, yes, but it'll depend on debate over "what should be in the XDR" among other topics. I suspect there'll be some code-level questions as well.

Reply:
There's actually a weird sort of nearly-circular dependency here. Only due to the fact that I'm changing both XDR and host crate here, but still: if I change the XDR in this PR then the generated XDR changes and the env crate changes and then this PR should be re-updated to contain an updated reference to the env crate. The cycle stops there, but .. it's a little awkward!

Reply:
Doh. I'm not sure how we get away from that. I mean, we sort of already have that today with xdrpp? Except that the generation of code occurs inside stellar-core.